Solidification/Melting Models for Navier Stokes (#23358)
idaholab/moose: modules/navier_stokes/doc/content/source/auxkernels/NSComputeLiquidFraction.md

# NSComputeLiquidFraction

!syntax description /AuxKernels/NSComputeLiquidFraction

The liquid fraction is defined as follows:

\begin{equation}
f_l =
\begin{cases}
f_l = 0, \text{ if } T < T_{solidus}, \\
f_l = \frac{T - T_{solidus}}{T_{liquidus} - T_{solidus},}, \text{ if } T \in (T_{solidus}, T_{liquidus}), \\
f_l = 1, \text{ if } T > T_{liquidus}, \\
\end{cases}
\end{equation}

> where $T_{liquidus}$ and $T_{solidus} < T_{liquidus}$ are the liquidus and solidus temperatures.

Reviewer: ```suggestion
where $T_{liquidus}$ and $T_{solidus}$ are the liquidus and solidus temperatures respectively; note that $T_{solidus}$ must be less than $T_{liquidus}$.
```

The liquidus and solidus temperature must be defined by the user.
The larger the difference between these two temperatures, the more stable the numerical behavior of the code but the more diffusive the solutions obtained.

!alert note

> There is no need to define the solid fraction ($f_s$), this one gets defined internally in the solidification objects as $1 - f_l$.

Reviewer: ```suggestion
There is no need to define the solid fraction ($f_s$); it is defined internally in the solidification objects as $1 - f_l$.
```

> # NSComputeLiquidFraction

Reviewer 2: Why is "compute" in the name? Seems unusual.
